src/ucp/wireup/wireup_cm.c

```diff
@@ -370,12 +370,11 @@ ucp_wireup_cm_ep_cleanup(ucp_ep_t *ucp_ep, ucs_queue_head_t *queue)
         /* During discarding transfer the pending queues content from the
          * previously configured UCP EP to a temporary queue for further
          * replaying */
-        ucp_worker_discard_uct_ep(ucp_ep, ucp_ep_get_lane(ucp_ep, lane_idx),
-                                  ucp_ep_get_rsc_index(ucp_ep, lane_idx),
-                                  UCT_FLUSH_FLAG_CANCEL,
-                                  ucp_request_purge_enqueue_cb, queue,
-                                  (ucp_send_nbx_callback_t)ucs_empty_function,
-                                  NULL);
+        ucp_worker_discard_uct_ep(
+                ucp_ep, ucp_ep_get_lane(ucp_ep, lane_idx),
+                ucp_ep_get_rsc_index(ucp_ep, lane_idx), UCT_FLUSH_FLAG_CANCEL,
+                (uct_pending_purge_callback_t)ucs_empty_function, NULL,
+                (ucp_send_nbx_callback_t)ucs_empty_function, NULL);
```

Comment:
do we expect that pending queues are empty? so can we pass `NULL` instead of `ucs_empty_function` in both places?

Reply:
In theory yes, you are right, but ucp_worker_discard_uct_ep has an assertion that checks if purge_cb != NULL

Reply:
then could pass `ucs_empty_function_do_assert_void`, WDYT?

Reply:
Hi.
- I'm not sure why you prefer ucs_empty_function_do_assert_void over `ucs_empty_function`, can you please explain?
- We have more places with calls to `ucp_worker_discard_uct_ep` that we pass `ucs_empty_function` as purge_cb

Reply:
> I'm not sure why you prefer `ucs_empty_function_do_assert_void` over `ucs_empty_function`, can you please explain?

to detect bugs earlier

> We have more places with calls to `ucp_worker_discard_uct_ep` that we pass `ucs_empty_function` as purge_cb, I think we need to be consistent.

We are discussing this particular case where expectation by design is to have queues empty. In other cases, probably, expectation is to do no-op on purging.

Reply:
I see. You are right.